src/main/java/randoop/generation/NonSutClassSet.java

```diff
@@ -0,0 +1,90 @@
+package randoop.generation;
+
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.regex.Pattern;
+import org.checkerframework.checker.signature.qual.ClassGetName;
+import randoop.main.GenInputsAbstract;
+import randoop.reflection.AccessibilityPredicate;
+
+/**
+ * Keep track of classes used during demand-driven input creation that are not part of the software
+ * under test (SUT), i.e., not explicitly specified by the user via {@code --classlist} or {@code
+ * --testjar}. This class maintains both all Non-SUT classes and Non-SUT classes that are not part
+ * of the JDK.
+ */
+public class NonSutClassSet {
+
+  /** The set of classes that are part of the SUT. */
+  private final Set<@ClassGetName String> sutClasses =
+      GenInputsAbstract.getClassnamesFromArgs(AccessibilityPredicate.IS_ANY);
+
+  /** The set of classes used during input creation that are not part of the SUT. */
+  private final Set<Class<?>> nonSutClasses;
+
+  /** Non-SUT classes that are not part of the JDK or primitive types. */
+  private final Set<Class<?>> nonJdkNonSutClasses;
+
+  /** Matches JDK classes (including array types like [Ljava.lang.String;). */
+  private static final Pattern JDK_CLASS_PATTERN = Pattern.compile("^(\\[+L)?java\\..");
+
+  /** Creates a NonSutClassSet. */
+  public NonSutClassSet() {
+    nonSutClasses = new LinkedHashSet<>();
+    nonJdkNonSutClasses = new LinkedHashSet<>();
+  }
+
+  /**
+   * Records a class as a non-SUT class.
+   *
+   * @param cls the class to record
+   */
+  public void addNonSutClass(Class<?> cls) {
+    nonSutClasses.add(cls);
+    String name = cls.getName();
+    if (!isJdkClass(name) && !cls.isPrimitive()) {
+      nonJdkNonSutClasses.add(cls);
+    }
+  }
+
+  /**
+   * Returns the set of classes that are part of the SUT.
+   *
+   * @return an unmodifiable set of all classes that are part of the SUT
+   */
+  public Set<@ClassGetName String> getSutClasses() {
+    return Collections.unmodifiableSet(new LinkedHashSet<>(sutClasses));
```

Comment:
There is no need to create a new `LinkedHashSet` every time this method is called.  `sutClasses` will never change, so you can set `sutClasses` to `Collections.unmodifiableSet(new LinkedHashSet<>(...)` and then return `sutClasses` directly.